tutorial: Migrate to Stargate Remove Launchpad as version name (#604)
cosmos/sdk-tutorials: launchpad-to-stargate/tutorial/01-introduction.md

order: 1
---

# Migrating from v0.39 to v0.40+ using Starport

> This tutorial will present a guide on how to migrate your Cosmos SDK application from v0.39 to v0.40+ "Stargate". The v0.40+ release comes with many features and enhancements, such as IBC, efficiency gains, protobuf support, and more.

Reviewer: hmmm don't we want to be equal in our mentions of launchpad and stargate?

curl https://get.starport.network/starport@v0.13.1! | bash
```

> We will be migrating the Proof of File existence tutorial from v0.39 to v0.40+ Stargate. At the time of writing, the version you migrate from is `v0.39.1` and Stargate is `v0.40.0` of the Cosmos SDK.

Author: We can still use Stargate, but we want to reserve the term "Launchpad" for something different. The release will probably still be named "Launchpad" but we try to avoid to use the term here.

> In order to obtain the code for the Proof of File Existence application, you can follow the tutorial [here](https://tutorials.cosmos.network/proof-of-file-existence/tutorial/01-intro.html), or clone [this](https://github.com/cosmos/sdk-tutorials/tree/master/proof-of-file-existence/pofe) repository.

Reviewer: ```suggestion
To get the code for the Proof of File Existence (PoFE) application, clone the [pofe repo](https://github.com/cosmos/sdk-tutorials/tree/master/proof-of-file-existence/pofe).
```

Reviewer: Tip: provide only one method to accomplish a task. I selected the repo clone method, since the linked topic didn't provide much useful information

## Resources

> Throughout this guide, we will be regularly referencing this [document](https://docs.cosmos.network/master/migrations/app_and_modules.html), as it contains crucial information about how to migrate a module and application from v0.39 to v0.40+ Stargate.

Reviewer: ```suggestion
This tutorial regularly references the [App and Modules Migration](https://docs.cosmos.network/master/migrations/app_and_modules.html) documentation that contains crucial information about how to migrate a module and application from v0.39 to v0.40 and later (Stargate). 
```

> As the design of Cosmos SDK application is meant to be modular and composable, it is possible to migrate an application from v0.39 to v0.40+ by copying the modules over to a boiler Stargate application and updating a few of the commands mentioned in the migration docs. We will be using Starport to assist with scaffolding new files which will be used as a baseline for migrating your application logic.

Reviewer: ```suggestion
The modular and composable design of Cosmos SDK applications simplifies the migration process by copying modules from v0.39 to a boilerplate Stargate v0.40 and later application and then updating the commands. This tutorial uses Starport to scaffold new files that are used as a baseline for migrating your application logic.
```

Reviewer: we can probably tighten even more, but I think this is a good start

## Boilerplate application

> Place the v0.39 PoFe application in a `launchpad` directory and next to it a `stargate` directory. We will aim at moving the logic from one application to the other and have your blockchain Stargate ready by the end of the tutorial.
> 
> The directory structure should look like this:

Reviewer: ```suggestion
In this tutorial, you move the logic from the `launchpad` application to the `stargate` application to make your blockchain Stargate-ready.

1. Put the v0.39 PoFE application in a `launchpad` directory.
1. Create an adjacent `stargate` directory. 

The directory structure should look like this:
```

Reviewer: on line 45, change to

> After your application is ready, use Starport to assist with the migration to Stargate.

(how do we know if the application is ready? because the folder structures are in place? if so let's say that instead)

Reviewer: on line 53, change to

> After the directory structure is in place, you have a folder called `pofe` with your boilerplate Stargate application.

> This includes folders such as `pofe/internal`, `pofe/scripts`, `pofe/third_party`, and `pofe/proto`, as well as files following the regex `*.pb.go` and `*.pb.gw.go`.

Reviewer: ```suggestion
The directory includes `pofe/internal`, `pofe/scripts`, `pofe/third_party`, and `pofe/proto` folders and the regex `*.pb.go` and `*.pb.gw.go` files.
```

> Now we have a better understanding of the differences in our application. In the next section, we will migrate our application from v0.39 to v0.40+ Stargate.

Reviewer: ```suggestion
Now you have a better understanding of the differences in your application and are prepared to migrate your application from v0.39 to v0.40 and later (Stargate).
```
